### DISCUSSION TITLE
Top Tip for users with 'stationary' MacBooks

### DISCUSSION THREAD

**Discussion by danfoerster** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#172] [Ideas]
This is a tip for users who have their Macbook plugged in most of the time and barely ever move it from their desks. This will essentially bypass the battery entirely and allow it to dwell in a SoC (state of charge) between 20 and 80% (or any value of your choosing) until the lower value is reached or the setting in AlDente is changed. This will also eliminate any 'top-off' charges that would otherwise try to maintain the battery at a certain level. Meaning, the battery can instead quietly rest and over the course of a month or even longer incredibly slowly and completely undisturbed drain from and to your chosen percentages, without receiving any current and is basically treated 'as if in storage' for this amount of time.
Since storing Lithium ion batteries poses virtually no problem as long as there is charge left in it and temperatures are in control, I think that it does quite the opposite of hurting it. For temperature-control of especially Intel MacBooks I'd highly recommend to download MacsFanControl alongside with it, if you don't already have it installed. Having the fans increase earlier is not only going to make your computer more comfortable to handle, but it also keeps the inside of the enclosure much more ventilated which means, the battery and frankly all other components sit in a cooler more suited environment. Goes without saying, don't keep your MacBook (or any electronic device with a battery) next to a radiator or have a radiator underneath your desk turned on all the time or leave your device cooking in direct sunlight. Because if you do, not even this phenomenal tool is going to magically save your battery.

The way of setting it up like this has the sole purpose of preventing the power adapter from re-charging the battery on a regular basis in order for it to stay on a certain percentage. If you don't need your Mac always ready to go with a charge of e.g. 80%, it is simply not necessary and actually counter intuitive.
Now, in order for this to work, your Mac should/must be left plugged in all the time and essentially be treated as if the battery wasn't even there. The charge light (if you have the Magsafe charger) will always be orange and power will always be drawn from the power adapter and never the battery.

Set AlDente to 76% (or any value) however 76 will result in the menu bar battery icon showing 80% at first (and because I like even numbers) and this number will **very** slowly go down as the battery is losing charge on its own. Ideally you would want to choose a value closer to 50%, but then again, this might be too little charge if by any chance you need the Mac ready to go. But you can also start from 100% (although that isn't advised, as even Apple states, one should charge it only to around 50% if it is planned that the battery will not be used for some time. Battery university advises that 40% is even better for storage). But then again, if you spontaneously should need to unplug your Mac, only having a charge of 50 or 40% might be too low for most users. It will however initially self-discharge/drain quickest from 100% to 80%, and then slower from 80 to 30 or so. However as I experience it so far, it could take months until the percentage drops down from e.g. 80 to 20% (as long as the Mac hasn't been unplugged and hasn't run on battery power for any period of time at some point). But the thing is, you can restart the procedure however often and whenever you want. I suppose its just easier and more convenient if it's a 'stationary' MacBook that doesn't get moved around and unplugged on a daily basis. Otherwise I guess, this is just too cumbersome for everyday use because you want/require a full charge when you unplug it and leave the house. But since the self-discharge rate of the battery itself is so minimal (if your battery generally is still healthy and provides decent performance, that is) it may even work for those that are very strict of having 100% displayed. And even if you do this, it could take literally days or even weeks for it to drop to 99%.

*If you have an M1 Mac and do not use the Classic SMC mode in AlDente you might can type in 80 directly.
**But you must uncheck 'Allow Discharge'!**

*80% is the top limit of the sweet spot of Lithium ion batteries, or so I have read. So it would be wise to start from there or preferably something closer to 50%.

After setting your charge limit, allow your Mac to discharge/charge it to that percentage until the lightning bolt in the battery menu bar icon turns into a cable with a wall connector. Then you will know your Mac is powered by the power adapter and not using the battery. It will also say so if you expand the stock battery panel in the menu bar.
Once your desired percentage is reached, go back into AlDente and decrease the value. I have mine set to 20% for now as I am still testing what works best and to push back the moment the charger starts to charge the battery again as far as possible (and because this is the lowest possible. I also would not necessarily want to drain below 20% if I didn't necessarily have to). You can choose any two values of percentages you want your Mac to dwell in between (80-20, 80-50, 70-40 etc). **(On Intel Macs, AlDente can be quit now as the value is written into the Mac's SMC and does not need AlDente to run for it to work. You must not quit it on M1 machines!)**

Now the battery is essentially allowed to calmly rest and very slowly self-discharge without any load from this point forward. Depending how old/worn your battery is, it might take a week or if it's relatively new or healthy even months to completely drain. (Granted, I did replace mine about 4 months ago, and it hasn't even lost 0,5% while self-discharging in 2 days. I checked it with coconutbattery. The Macbook was in standby over night and plugged in all this time). FYI, if your menu bar shows 100%, it will do so, long after the actual, more accurate percentage has already dropped to around 95%. Only then the menu bar % will drop to 99.

I think its an interesting approach to set it up like this, if your Mac is sitting on a desk all the time and constantly plugged in anyway (maybe in clamshell mode with an external display). It will prevent the frequent top-off, because as I see it, there's no need to always keep the Mac at 80% if its always plugged in and drawing power from the wall anyways and you are not intending to go anywhere where you would have to bring it. If you do know you'll need your Mac charged, just open Al-Dente again, change the value to 80 (or whatever) again and let your MacBook charge. Hopefully they implement a 'Fully charge now' button and other features in the next version, as this would make this even easier. Presets maybe even. Maybe a 'Clamshell mode' that does exactly what I have been trying to describe here :D

Please someone share their 2cents or correct me if this is utter nonsense. I am curious to know what you think about this!

**Comment by MatthiasKerbl** [comment]
Hi @danfoerster,

thank you very much for this detailed description of how to minimally use our MacBook Batteries with AlDente 2.1. We will definitely implement a feature in this regard in our upcoming AlDente Pro.

Best regards,
Matthias

**Reply by mikebore** [reply]
Thanks Dan, all makes sense. I was beginning to think along the same lines while doing the testing with "Allow Discharge" off in this post https://github.com/davidwernhart/AlDente/discussions/169#discussioncomment-449822 and noticing that the battery charge was dropping extremely slowly because the machine was basically running on the charger. 
You have taken the thinking to the next logical point.

**Reply by mikebore** [reply]
A few more thoughts. 

I have started doing your proposal with Mac on 80% charge, and I have changed AlDente to 40%, with 'Allow Discharge' off. As I understand your proposal it will now slowly self discharge to 40% which will take about a "week". Then I should change AlDente to 80% to charge it up, then back to 40% and so on, with each cycle taking probably a week or so. Apart from the regular settings changes, a possible drawback is that this will add a cycle to the cycle count each "week". Maybe the "week" is month, maybe two days...will find out.

The main benefit is that this prevents the frequent top off to keep it at 80% to replace the slow self discharge.

But how frequent are the top offs, which only take a few minutes? and are they damaging to battery health? Does the battery health prefer a slow discharge/recharge over a "week", or more frequent short top offs....which will not add to the cycle count? I don't know the answers.

A slightly modified approach would be for AlDente to allow us to specify a band within which to stay, which would avoid the frequent AlDente settings changes and reduce the frequency of top-offs, without increasing cycle count. EG if the band was 80-75 it would self discharge from 80 down to 75 then be automatically topped up to 80. This would not increase the cycle count while reducing the top off frequency.

A lot of speculation here and I don't know what effects the options would have long term.

**Reply by danfoerster** [reply]
Hello! How long it takes for your battery to reach the 40% is depending on how 'good' your battery is still. Could take a week, could also take a month or even more. For example. I charged my Mac to 80% on Saturday and did not once unplugged it since then. Today is Tuesday. So roughly 4 days without ever actively utilizing the battery. AlDente was set to 76, Coconutbattery showed a SoC of 76,8% when the charging stopped. Then I set AlDente to 20.
Now, I am down to 76,2%. So basically 0,6% in 4 days. So lets say, for my battery, at 80% SoC, the self-discharge rate of one week is ~1%. (Keep in mind, my battery is virtually new with 117% health). Do the math how long it would take to drain all the way to empty :p If I would never unplug my Mac, and it would keep discharging at this rate, it would take a good year to drain it to 20 or 30%. So essentially, this would mean that after draining and recharging I would have only gained one charge cycle in one year. Of course that is unfeasible because I will unplug my Mac in the meantime, and use the battery etc. but in theory it would be possible I think.

But I think it makes no difference to the battery itself whether it gets discharged by actual usage or wether it is just left to self-discharge over an extended period of time. What I could research, the only bad thing that degrades a Lithium ion battery in storage (which is essentially what we would be doing here) is if it is completely empty (fully charged is also not ideal) or if it is stored in temperates above 35°C.

I think it depends on what percentage the top-offs occur. The higher the %, the lower the Watts but the higher the voltage.
A top-off from 49 to 50% is much better and less strenuous I think, than a top-off from e.g. 99 to 100% because it needs to push the max voltage for the cell to completely charge up. Thats the only reason why its bad to keep a battery at 100%. Because it always has to push in the max voltage in order to top it off.

But if I am now thinking about it, if the Mac takes about one or two weeks to drop from 80% to 79% and then tops off, this is also not bad at all. One top-off charge in 2 weeks? Not harmful at all. But of course it depends how one uses his Mac and how often it gets unplugged (even if it just for a couple minutes). Because if the battery drains a percent in 10 minutes, after plugging back in it will force a top-off even if its not needed. That is my basic peeve. I don't want my Macbook to always top off because it lost one lousy percent. And frankly, it doesn't have to. At all.

**Reply by mikebore** [reply]
Thanks, very interesting. Yes the "week" to self discharge from 80 to 40 might well be very much longer, in which case both the possibility of higher cycle count and frequent setting changes are much less important.

As you say the subtleties about size and frequency of top offs are probably much less important than avoiding 100% and full discharge states.
